### ISSUE TITLE
model weight small after prompt tuning

### ISSUE THREAD

**Issue by ccsvd** [microsoft/GLIP#114]
when i promt tuning the weight file reduced from 3.5GB to 800M. Is that normal?

**Comment by Wangman1** [comment]
Hello, may I ask you to solve this problem? I have also met, and the AP will be reduced after prompt tuning/fine-tuning

**Comment by ccsvd** [comment]
这是来自QQ邮箱的假期自动回复邮件。 您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by ccsvd** [comment]
yes，i find the weight small is the download model include a ema model weight, but when prompt tuning it not use ema train, you can check.
when prompt tuning, the ap can increase, it can work, may be you can check your train process.

**Comment by Wangman1** [comment]
非常感谢您的回复，能看看您的 prompt tuning coco 的命令吗，我按作者提供的命令来 prompt tuning 的话 AP 会一直下降

**Comment by ccsvd** [comment]
CUDA_VISIBLE_DEVICES=2 python -m torch.distributed.launch --nproc_per_node=1 --master_port 29502 tools/finetune.py \
      --config-file configs/pretrain/glip_Swin_T_O365_GoldG.yaml \
      --ft-tasks configs/your.yaml \
      --skip-test \
      --custom_shot_and_epoch_and_general_copy 0_200_1 \
      --evaluate_only_best_on_test --push_both_val_and_test \
      MODEL.WEIGHT MODEL/glip_tiny_model_o365_goldg_cc_sbu_lvisbest.pth MODEL.BACKBONE.FREEZE_CONV_BODY_AT 2 MODEL.DYHEAD.USE_CHECKPOINT True \
      DATASETS.TRAIN_DATASETNAME_SUFFIX _grounding DATASETS.USE_OVERRIDE_CATEGORY True DATASETS.SHUFFLE_SEED 3 DATASETS.USE_CAPTION_PROMPT True DATASETS.DISABLE_SHUFFLE True \
      TEST.DURING_TRAINING True TEST.IMS_PER_BATCH 6 TEST.EVAL_TASK detection \
      SOLVER.USE_AMP True SOLVER.IMS_PER_BATCH 6 SOLVER.FIND_UNUSED_PARAMETERS False SOLVER.TEST_WITH_INFERENCE True SOLVER.USE_AUTOSTEP True  SOLVER.SEED 10 \
      SOLVER.STEP_PATIENCE 2 SOLVER.CHECKPOINT_PER_EPOCH 1.0 SOLVER.AUTO_TERMINATE_PATIENCE 4 \
      SOLVER.MODEL_EMA 0.0 \
      SOLVER.WEIGHT_DECAY 0.65 \
      SOLVER.BASE_LR 0.1 \
      SOLVER.TUNING_HIGHLEVEL_OVERRIDE language_prompt_v2

**Comment by Wangman1** [comment]
谢谢谢谢谢谢~~~

**Comment by zilong69** [comment]
> CUDA_VISIBLE_DEVICES=2 python -m torch.distributed.launch --nproc_per_node=1 --master_port 29502 tools/finetune.py --config-file configs/pretrain/glip_Swin_T_O365_GoldG.yaml --ft-tasks configs/your.yaml --skip -test --custom_shot_and_epoch_and_general_copy 0_200_1 --evaluate_only_best_on_test --push_both_val_and_test MODEL.WEIGHT MODEL/glip_tiny_model_o365_goldg_cc_sbu_lvisbest.pth MODEL.BACKBONE.FREEZE_CONV_BODY_AT 2 MODEL.DYHEAD.USE_CHECKPO INT True DATASETS.TRAIN_DATASETNAME_SUFFIX _grounding DATASETS.USE_OVERRIDE_CATEGORY True DATASETS.SHUFFLE_SEED 3 DATASETS.USE_CAPTION_PROMPT True DATASETS。 DISABLE_SHUFFLE True TEST.DURING_TRAINING True TEST.IMS_PER_BATCH 6 TEST.EVAL_TASK 检测 SOLVER.USE_AMP 正确 SOLVER.IMS_PER_BATCH 6 SOLVER.FIND_UNUSED_PARAMETERS 错误 SOLVER.TEST_WITH_INFERENCE 正确 SOLVER.USE_AUTOSTEP 正确 SOLVER.SEED 10 SOLVER.STEP_PATIENCE 2 SOLVER.CHECKPOINT_PER_EPOCH 1.0 SOLVER.AUTO_TERMINATE_PATIENCE 4 SOLVER.MODEL_ EMA 0.0 求解器 .WEIGHT_DECAY 0.65 求解器 .BASE_LR 0.1 求解器。 TUNING_HIGHLEVEL_OVERRIDE language_prompt_v2

**Comment by zilong69** [comment]
> CUDA_VISIBLE_DEVICES=2 python -m torch.distributed.launch --nproc_per_node=1 --master_port 29502 tools/finetune.py --config-file configs/pretrain/glip_Swin_T_O365_GoldG.yaml --ft-tasks configs/your.yaml --skip -test --custom_shot_and_epoch_and_general_copy 0_200_1 --evaluate_only_best_on_test --push_both_val_and_test MODEL.WEIGHT MODEL/glip_tiny_model_o365_goldg_cc_sbu_lvisbest.pth MODEL.BACKBONE.FREEZE_CONV_BODY_AT 2 MODEL.DYHEAD.USE_CHECKPO INT True DATASETS.TRAIN_DATASETNAME_SUFFIX _grounding DATASETS.USE_OVERRIDE_CATEGORY True DATASETS.SHUFFLE_SEED 3 DATASETS.USE_CAPTION_PROMPT True DATASETS DISABLE_SHUFFLE True TEST.DURING_TRAINING True TEST.IMS_PER_BATCH 6 TEST.EVAL_TASK 检测 SOLVER.USE_AMP 正确 SOLVER.IMS_PER_BATCH 6 SOLVER.FIND_UNUSED_PARAMETERS 错误 SOLVER.TEST_WITH_INFERENCE 正确 SOLVER.USE_AUTOSTEP 正确 SOLVER.SEED 10 SOLVER.STEP_PATIENCE 2 SOLVER.CHECKPOINT_PER_EPOCH 1.0 SOLVER.AUTO_TERMINATE_PATIENCE 4 SOLVER。 MODEL_ EMA 0.0 活动器 .WEIGHT_DECAY 0.65 活动器 .BASE_LR 0.1 启动器。 TUNING_HIGHLEVEL_OVERRIDE language_prompt_v2

Hello, could you please take a look at your. ymal file

**Comment by ccsvd** [comment]
这是来自QQ邮箱的假期自动回复邮件。 您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by glevin86** [comment]
I am also getting the AP reduced after finetuning, with this command: Why?

python tools/finetune.py --config-file configs/pretrain/glip_Swin_L.yaml --out_dir output_fine_tune/only_ambulance/ --ft-tasks configs/MBE/mbe_config.yaml --skip-test --custom_shot_and_epoch_and_general_copy 1_200_8 --evaluate_only_best_on_test --push_both_val_and_test MODEL.WEIGHT glip_large_model.pth SOLVER.USE_AMP True TEST.DURING_TRAINING True TEST.IMS_PER_BATCH 1 SOLVER.IMS_PER_BATCH 1 SOLVER.WEIGHT_DECAY 0.05 TEST.EVAL_TASK detection DATASETS.TRAIN_DATASETNAME_SUFFIX _grounding MODEL.BACKBONE.FREEZE_CONV_BODY_AT 2 MODEL.DYHEAD.USE_CHECKPOINT True SOLVER.FIND_UNUSED_PARAMETERS False SOLVER.TEST_WITH_INFERENCE True SOLVER.USE_AUTOSTEP True DATASETS.USE_OVERRIDE_CATEGORY True SOLVER.SEED 10 DATASETS.SHUFFLE_SEED 3 DATASETS.USE_CAPTION_PROMPT True DATASETS.DISABLE_SHUFFLE True SOLVER.STEP_PATIENCE 3 SOLVER.CHECKPOINT_PER_EPOCH 1.0 SOLVER.AUTO_TERMINATE_PATIENCE 8 SOLVER.MODEL_EMA 0.0 SOLVER.TUNING_HIGHLEVEL_OVERRIDE full

**Comment by ccsvd** [comment]
这是来自QQ邮箱的假期自动回复邮件。 您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
